service/src/main/java/bio/terra/landingzone/common/utils/MetricUtils.java

```diff
@@ -0,0 +1,63 @@
+package bio.terra.landingzone.common.utils;
+
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Timer;
+import java.time.Duration;
+
+public class MetricUtils {
+  private static final String NAMESPACE = "landingzone";
+  private static final String CLOUD_PLATFORM_TAG = "cloudPlatform";
+  private static final String AZURE_PLATFORM_NAME = "AZURE";
+  private static final String LANDINGZONE_TYPE_TAG = "type";
+  private static final String LANDINGZONE_STEP_TAG = "step";
+
+  // this is to set reasonable upper/lower bound for a step. postgres, aks take 7-8 minutes;
+  // histogram metric would have buckets for all results.
+  private static final int LANDINGZONE_STEP_MAX_DURATION_MILLISECONDS = 60 * 10 * 1000; // 10min
+  private static final int LANDINGZONE_STEP_MIN_DURATION_MILLISECONDS = 60 * 1000; // 1min
+
+  private MetricUtils() {}
+
+  public static void incrementLandingZoneCreation(String type) {
```

Comment:
I think there should be a `creation.start` metric and also a `creation.success` so we're symmetric w/the failure metric.

Reply:
We have 2 metrics now: 1) counts all request 2) counts failed requests. Not sure I understand meaning of 'creation.start'. I will add comments.

Reply:
Idea with counter now is to measure total number of landing zone creation request and also number of requests which failed.

Reply:
Ok, can we count the successful creation of landing zones? Or do we not have a good way to do that?

Reply:
Put another way: a possible metric hierarchy would look like this, if we had all the hooks we need:

1. Total LZ creation requests
2. Total LZ creation successed
3. Total LZ creation failures
4. Associated timers

How does the namespace get organized given the above?

Reply:
so, within current implementation we have:

1. `landingzone.creation.count` for total number of requests
2. we don't have metric for 'total' number of successful request, but we can count it based on 1,3
3. `landingzone.creation.failure.count` for total number of failed requests
4. `landingzone.step.latency` metric to measure step duration (here we measure duration only for the Azure resource flight steps). Steps to create SAM resource and checking billing profile existence are ignored since their duration is really small comparing to Azure resource creation.

We could probably count success based on the success of a latest step. But I am not sure how the future flight might look like for different type of landing zones. Maybe we would have different final steps for a different type of landing zones.